File: Examples/Perl/Fields/AddFormFields.pl
#ExStart:1
use lib 'lib';
use strict;
use warnings;
use utf8;
use File::Slurp; # From CPAN
use JSON;

use AsposeStorageCloud::StorageApi;
use AsposeStorageCloud::ApiClient;
use AsposeStorageCloud::Configuration;
use AsposeWordsCloud::WordsApi;
use AsposeWordsCloud::ApiClient;
use AsposeWordsCloud::Configuration;
use AsposeWordsCloud::Object::BookmarkData;

my $configFile = '../Config/config.json';
my $configPropsText = read_file($configFile);
my $configProps = decode_json($configPropsText);
my $data_path = '../../../Data/';
my $out_path = $configProps->{'out_folder'};

$AsposeWordsCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposeWordsCloud::Configuration::api_key = $configProps->{'api_key'};
$AsposeWordsCloud::Configuration::debug = 1;
$AsposeStorageCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposeStorageCloud::Configuration::api_key = $configProps->{'api_key'};

# Instantiate Aspose.Storage and Aspose.Words API SDK
my $storageApi = AsposeStorageCloud::StorageApi->new();
my $wordsApi = AsposeWordsCloud::WordsApi->new();

# Set input file name
my $name = 'SampleBlankWordDocument.docx';
my $sectionIndex = 0;
my $paragraphIndex = 0;
my $insertBeforeNode = "";

my $xmlBody = '<FormFieldTextInput>'
            . '<Name>MyName</Name>'
            . '<Enabled>true</Enabled>'
            . '<StatusText />'
            . '<OwnStatus>false</OwnStatus>'
            . '<HelpText />'
            . '<OwnHelp>false</OwnHelp>'
            . '<CalculateOnExit>true</CalculateOnExit>'
            . '<EntryMacro />'
            . '<ExitMacro />'
            . '<TextInputFormat>UPPERCASE</TextInputFormat>'
            . '<TextInputType>Regular</TextInputType>'
            . '<TextInputDefault>Farooq Sheikh</TextInputDefault>'
            . '</FormFieldTextInput>';

# Upload file to aspose cloud storage 
my $response = $storageApi->PutCreate(Path => $name, file => $data_path.$name);

my $bookmarkDataBody = AsposeWordsCloud::Object::BookmarkData->new( 'Name' => 'aspose', 'Text' => 'Bookmark is very good');

# Invoke Aspose.Words Cloud SDK API to add field in the document     
	$response = $wordsApi->PutFormField(name=> $name, sectionIndex=>$sectionIndex, 
										paragraphIndex=>$paragraphIndex, insertBeforeNode=>$insertBeforeNode, body=>$xmlBody);

if($response->{'Status'} eq 'OK'){
	my $bookmark = $response->{'Bookmark'};
	# Download updated document from storage server
	$response = $storageApi->GetDownload(Path => $name);
	my $output_file = 'updated-' . $name;	
	write_file($out_path.$output_file, { binmode => ":raw" }, $response->{'Content'});
	}
#ExEnd:1

File: Examples/Node.js/Fields/AddFormFields.js
// ExStart:1
var fs = require('fs');
var assert = require('assert');
var StorageApi = require('asposestoragecloud');
var WordsApi = require('asposewordscloud');
var configProps = require('../Config/config.json');
var data_path = '../../../Data/';

var AppSID = configProps.app_sid;
var AppKey = configProps.api_key;
var config = {'appSid':AppSID,'apiKey':AppKey , 'debug' : true};

// Instantiate Aspose Storage API SDK
var storageApi = new StorageApi(config);
// Instantiate Aspose Words API SDK
var wordsApi = new WordsApi(config);

// Set input file name
var name = "SampleBlankWordDocument.docx";
var sectionIndex = 0;
var paragraphIndex = 0;

var formFieldBody = {
		'Name' : 'MyName',
		'Enabled' : 'true',
		'OwnStatus' : 'false',
		'CalculateOnExit' : 'true',
		'TextInputFormat' : 'UPPERCASE',
		'TextInputType' : 'Regular',
		'TextInputDefault' : 'Farooq Sheikh'
			};

try {
// Upload source file to aspose cloud storage
storageApi.PutCreate(name, versionId=null, storage=null, file= data_path + name , function(responseMessage) {

	assert.equal(responseMessage.status, 'OK');

	// Invoke Aspose.Words Cloud SDK API to add field in the document
	wordsApi.PutFormField(name, sectionIndex, paragraphIndex, null, null, null, null, formFieldBody, function(responseMessage) {
			assert.equal(responseMessage.status, 'OK');
			
			// Download updated document from storage server
			storageApi.GetDownload(name, null, null, function(responseMessage) {
				assert.equal(responseMessage.status, 'OK');
				var outfilename = 'updated-' + name;
				var writeStream = fs.createWriteStream(data_path + outfilename);
				writeStream.write(responseMessage.body);
				});
			});
	});

}catch (e) {
  console.log("exception in example");
  console.log(e);
}
//ExEnd:1